src/pygeoprocessing/geoprocessing.py

```diff
@@ -1630,8 +1630,18 @@ def reproject_vector(
     # Get the SR of the original_layer to use in transforming
     base_sr = layer.GetSpatialRef()
 
-    # Create a coordinate transformation
-    coord_trans = osr.CoordinateTransformation(base_sr, target_sr)
+    # Create a coordinate transformation [GDAL 2 NOT supported]
+    # In GDAL 3.0 spatial references have been updated. One difference is 
+    # how transforms work with lat/lon coordinate systems. Transforms
+    # expect (lat, lon) order in GDAL 3.0 as opposed to thge GIS friendly
+    # (lon, lat). Thus if using a (lat, lon) srs set for backwards 
+    # compatability. 
+    if not base_sr.IsProjected():
+        base_sr.SetAxisMappingStrategy(osr.OAMS_TRADITIONAL_GIS_ORDER)
+    if not target_sr.IsProjected():
+        target_sr.SetAxisMappingStrategy(osr.OAMS_TRADITIONAL_GIS_ORDER)
+        
+    coord_trans = osr.CreateCoordinateTransformation(base_sr, target_sr)
```

Comment:
But do should keep `CreateCoordinateTransformation` because that's a GDAL 3.0 thing!